### DISCUSSION TITLE
Phalcon 3 MongoDB ODM transactions

### DISCUSSION THREAD

**Discussion by jsakars** [phalcon/cphalcon#15695] [Q&A]
We are unable to figure out how can we use MongoDB transactions with Phalcon 3. Documentation is all around `db` which is clearly ORM e.g. MySQL related.

We are obviously using MongoDB version 4 where transaction support is available.

Could someone please enlighten us!?

**Comment by quasipickle** [comment]
v4 has no MongoDB support: https://docs.phalcon.io/4.0/en/db-odm

But there's no reason you can't use another library: https://docs.mongodb.com/drivers/php-libraries/

**Reply by jsakars** [reply]
I was referring to Phalcon version 3 which does have MongoDB support => https://docs.phalcon.io/3.4/en/db-odm

**Reply by quasipickle** [reply]
I'm not sure if you updated your post to specify version 3.  If you did - good move.  If you didn't (and it always said 3), then I need to learn to read better.

**Reply by jsakars** [reply]
> I'm not sure if you updated your post to specify version 3. If you did - good move. If you didn't (and it always said 3), then I need to learn to read better.

No, I did not :) You would see something like `edited at ...` in such case I believe :)

**Comment by Jeckerson** [comment]
@quasipickle Actually it has - https://github.com/phalcon/incubator-mongodb
Just docs need to be updated.

```
composer require phalcon/incubator-mongodb
```

**Reply by quasipickle** [reply]
I wasn't aware that stuff in an incubator was considered official - good to know.
